### PR TITLE
Doxygen conversation starter

### PR DESCRIPTION
**Pull request by gheber**
- Fixed dependencies (Doxygen, Graphviz) in Docker images
- Added `HERMES_ENABLE_DOXYGEN` option (`OFF` by default) and `dox` target
- Enable and create Doxygen HTML documentation via `cmake --build . --target dox`
- Open `.../build/_build/html/index.html` and enjoy the show!

### CODE REVIEW

File: doc/Doxyfile.in [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+WARN_IF_UNDOCUMENTED   = YES
+WARN_IF_DOC_ERROR      = YES
+WARN_NO_PARAMDOC       = NO
+WARN_FORMAT            = "$file:$line: $text"
+WARN_LOGFILE           =
+#---------------------------------------------------------------------------
+# Configuration options related to the input files
+#---------------------------------------------------------------------------
+INPUT                  = @PROJECT_SOURCE_DIR@/src \
+                         @PROJECT_SOURCE_DIR@/docs
```
Comment (ChristopherHogan) [inline doc/Doxyfile.in:111]: This file's directory is named `doc` instead of `docs`. Should they be the same?

**Comment by ChristopherHogan** [comment]
Looks like a good start. I think we will want to split the documentation into a public version and a developer version. The public version will be everything in `src/api`, and the developer version will be everything else. Do you know of a good way to do that besides having two separate `Doxyfile`s?

**Review by ChristopherHogan — APPROVED** [review]

File: doc/Doxyfile.in (hunk 1 above)
Comment (jya-kmu) [reply doc/Doxyfile.in:111]: Nice catch.

**Review by jya-kmu — APPROVED** [review]

File: doc/Doxyfile.in (hunk 1 above)
Comment (gheber) [reply doc/Doxyfile.in:111]: Yes.

**Comment by gheber** [comment]
https://docs.hdfgroup.org/hermes/index.html

**Pull request merged**